### PR TITLE
Add support for above/below fill/hatch to Slope

### LINKED ISSUE #9194: [FEATURE] Add fill properties to slope
It'd be useful to have an annotation like [BoxAnnotation](https://bokeh.pydata.org/en/latest/docs/user_guide/annotations.html#box-annotations) that creates a dividing line, but which can be angled. It's possible to achieve something similar with an image, but then interaction is extremely slow as the plot has to re-render the image at every frame.

#### Using [bands](https://bokeh.pydata.org/en/latest/docs/user_guide/annotations.html#bands)

This may be possible if we expand the features of the `Band` annotation. Currently, leaving `lower` or `upper` unspecified results in a blank figure, but we could change the behavior such that if either is unspecified the band continuous off infinitely.

I'd like to hear the thoughts of those more familiar with the capabilities of annotations.

### PR DESCRIPTION
**Pull request by mattpap**
image

fixes #9194

### CODE REVIEW

File: bokehjs/test/integration/features.ts [hunk 1]
```diff
@@ -137,4 +139,40 @@ describe("Feature", () => {
...
+      p.add_layout(slope)
+      return p
+    }
+
+    it("should allow above and below fill and hover in Slope annotation", async () => {
+      const gp = new GridPlot({
+        toolbar_location: null,
+        children: [
+          [plot(0.5, 10), 0, 0], [plot(-0.5, 10), 0, 1], [plot(0.5, 10, [-10, 15]), 0, 2], [plot(-0.5, 10, [-20, 5]), 0, 3],
+          [plot(0, 10), 1, 0], [plot(-1, -10), 1, 1], [plot(0.5, 10, [15, 40]), 1, 2], [plot(-0.5, 10, [5, 30]), 1, 3],
```
Comment (bryevdv) [inline bokehjs/test/integration/features.ts]: Missing edge cases for when the slope is entirely off-screen, and inverted y-axes.

File: examples/basic/annotations/slope.py [hunk 2]
```diff
@@ -16,7 +16,10 @@
 p.circle(xpts, ypts, size=6, alpha=0.6, fill_color=None)
 
 slope = Slope(gradient=slope, y_intercept=intercept,
-              line_color='orange', line_dash='dashed', line_width=4)
+              line_color='orange', line_dash='dashed', line_width=4,
+              above_fill_color='orange',
+              below_fill_color='red',
+              )
```
Comment (bryevdv) [inline examples/basic/annotations/slope.py]: There is not a single incidence of this formatting anywhere to my knowledge. For examples especially this is unacceptable, as vertical space is typically what is at a premium in displayed docs. 
```suggestion
              line_color='orange', line_dash='dashed', line_width=4,
              above_fill_color='orange', below_fill_color='red')
```

File: src/bokeh/models/annotations/geometry.py [hunk 3]
```diff
@@ -249,6 +249,28 @@ def __init__(self, *args, **kwargs) -> None:
...
+    above_hatch_props = Include(ScalarHatchProps, prefix="above", help="""
+    The {prop} values for the area above the line.
+    """)
+
+    below_fill_props = Include(ScalarFillProps, prefix="below", help="""
+    The {prop} values for the area below the line.
+    """)
+
+    below_hatch_props = Include(ScalarHatchProps, prefix="below", help="""
+    The {prop} values for the area below the line.
```
Comment (bryevdv) [inline src/bokeh/models/annotations/geometry.py:265]: These could clarify what happens for inverted axes

File: tests/unit/bokeh/models/_util_models.py [hunk 4]
```diff
@@ -41,6 +41,12 @@
 TEXT  = ["text_font", "text_font_size", "text_font_style", "text_color", "text_outline_color",
          "text_alpha", "text_align", "text_baseline", "text_line_height"]
 
+ABOVE_FILL  = [ f"above_{name}" for name in FILL ]
+ABOVE_HATCH = [ f"above_{name}" for name in HATCH ]
+
+BELOW_FILL  = [ f"below_{name}" for name in FILL ]
+BELOW_HATCH = [ f"below_{name}" for name in HATCH ]
```
Comment (bryevdv) [inline tests/unit/bokeh/models/_util_models.py]: formatting consistency
```suggestion
ABOVE_FILL  = [f"above_{name}" for name in FILL]
ABOVE_HATCH = [f"above_{name}" for name in HATCH]

BELOW_FILL  = [f"below_{name}" for name in FILL]
BELOW_HATCH = [f"below_{name}" for name in HATCH]
```

File: examples/basic/annotations/slope.py (hunk 2 above)
Comment (bryevdv) [reply examples/basic/annotations/slope.py]: Actually, please revert these changes entirely. Keeping them would require updating the gallery thumbnail. I prefer the clean look of the current example. If you want to have a specific example for [inside the user guide](https://docs.bokeh.org/en/latest/docs/user_guide/basic/annotations.html#slopes) then adding a completely new `slope_fill.py` example for that purpose would be acceptable.

**Review by bryevdv — CHANGES_REQUESTED** [review]
See comments inline

File: examples/basic/annotations/slope.py (hunk 2 above)
Comment (mattpap) [reply examples/basic/annotations/slope.py]: That was committed accidentally.

**Review by bryevdv — APPROVED** [review]

File: src/bokeh/models/annotations/geometry.py (hunk 3 above)
Comment (mattpap) [reply src/bokeh/models/annotations/geometry.py:265]: This would be better to explain in a common location for all those properties, possibly user guide.
